cmd/skaffold/app/cmd/run_test.go

```diff
@@ -37,3 +43,60 @@ func TestNewCmdRun(t *testing.T) {
 		t.CheckDeepEqual(false, opts.EnableRPC)
 	})
 }
+
+type mockRunRunner struct {
+	runner.Runner
+	artifactImageNames []string
+}
+
+func (r *mockRunRunner) BuildAndTest(_ context.Context, _ io.Writer, artifacts []*latest.Artifact) ([]build.Artifact, error) {
+	var result []build.Artifact
+	for _, artifact := range artifacts {
+		imageName := artifact.ImageName
+		r.artifactImageNames = append(r.artifactImageNames, imageName)
+		result = append(result, build.Artifact{
+			ImageName: imageName,
+		})
+	}
+
+	return result, nil
+}
+
+func (r *mockRunRunner) DeployAndLog(context.Context, io.Writer, []build.Artifact) error {
+	return nil
+}
+
+func TestBuildImageFlag(t *testing.T) {
+	testutil.Run(t, "", func(t *testutil.T) {
+		mockRunner := &mockRunRunner{}
+		t.Override(&createRunner, func(config.SkaffoldOptions) (runner.Runner, *latest.SkaffoldConfig, error) {
+			return mockRunner, &latest.SkaffoldConfig{
+				Pipeline: latest.Pipeline{
+					Build: latest.BuildConfig{
+						Artifacts: []*latest.Artifact{
+							{
+								ImageName: "first",
+							},
+							{
+								ImageName: "second-test",
+							},
+							{
+								ImageName: "test",
+							},
+							{
+								ImageName: "aaabbbccc",
+							},
```

Comment:
These take up a lot of space!

```suggestion
							{ImageName: "first"},
							{ImageName: "second-test"},
							{ImageName: "test"},
							{ImageName: "aaabbbccc"},
```